src/solver.cpp

```diff
@@ -521,20 +546,24 @@ SensitivityMethod Solver::getSensitivityMethodPreequilibration() const { return
 void Solver::setSensitivityMethod(const SensitivityMethod sensi_meth) {
     if (sensi_meth != this->sensi_meth)
         resetMutableMemory(nx(), nplist(), nquad());
-    sensitivityMethod(&this->sensi_meth, sensi_meth);
+    sensitivityMethod(sensi_meth, false);
 }
 
 void Solver::setSensitivityMethodPreequilibration(const SensitivityMethod sensi_meth_preeq) {
-    sensitivityMethod(&this->sensi_meth_preeq, sensi_meth_preeq);
+    sensitivityMethod(sensi_meth_preeq, true);
 }
 
-void Solver::sensitivityMethod(SensitivityMethod *member,
```

Comment:
Hope this recoding is better now than the fully abstract version I implemented before.

Reply:
Definitely better, would have kept the assignment out of the function though and called the function `checkSensitivityMethod`. Should be overall less code, right?